tests/components/srp_energy/test_init.py

```diff
@@ -0,0 +1,26 @@
+"""Tests for Srp Energy component Init."""
+from homeassistant.components import srp_energy
+
+from tests.components.srp_energy import init_integration
+
+
+async def test_setup_entry(hass):
+    """Test setup entry fails if deCONZ is not available."""
+    config_entry = await init_integration(hass)
+    assert config_entry.state == "loaded"
+    assert hass.data[srp_energy.SRP_ENERGY_DOMAIN]
+
+
+async def test_unload_entry(hass):
+    """Test being able to unload an entry."""
+    config_entry = await init_integration(hass)
+    assert hass.data[srp_energy.SRP_ENERGY_DOMAIN]
+
+    assert await srp_energy.async_unload_entry(hass, config_entry)
+    assert not hass.data[srp_energy.SRP_ENERGY_DOMAIN]
+
+
+async def test_async_setup_entry_with_exception(hass):
+    """Test exception when SrpClient can't load."""
+    await init_integration(hass, side_effect=Exception())
+    assert srp_energy.SRP_ENERGY_DOMAIN not in hass.data
```

Comment:
Assert the config entry state instead. It should be `ENTRY_STATE_SETUP_ERROR` if the integration setup fails and `ENTRY_STATE_SETUP_RETRY` if it retries.